### PR TITLE
fix: add none value in document construction

### PR DESCRIPTION
**Pull request by dongxiang123**
Goals:

The None values in the column will cause errors in document construction.
- ...The following snippet can be executed

```python
from docarray import Document, DocumentArray

DA_CONFIG = {
    "storage": "weaviate",
    "config": {
        "name": "Persisted",
        "host": "localhost",
        "port": 8080,
        "vector_cache_max_objects": 3000000,
        "columns": [("rank", "int"), ("year", "int")]
    }
}

da = DocumentArray(**DA_CONFIG)
docs = [Document(text="Some text here", rank=2, year=2020),
        Document(text="Another kind of text here", rank=1, year=2022),
        Document(text="This document is bad", rank=2, year=3)]

with da:
    da.clear()
    da.extend(docs)

da[:,'tags__rank']
But when we have a None value then we can't index to a Document containing a None in a column that was specified with a certain datatype.

from docarray import Document, DocumentArray

DA_CONFIG = {
    "storage": "weaviate",
    "config": {
        "name": "Persisted",
        "host": "localhost",
        "port": 8080,
        "vector_cache_max_objects": 3000000,
        "columns": [("rank", "int"), ("year", "int")]
    }
}

da = DocumentArray(**DA_CONFIG)
docs = [Document(text="Some text here", rank=2, year=2020),
        Document(text="Another kind of text here", rank=1, year=2022),
        Document(text="This document is bad", rank=None, year=3)]

with da:
    da.clear()
    da.extend(docs)
```
# this will break
da[:,'tags__rank']
- ...
- [ ] check and update documentation, if required. See [guide](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#documentation-guidelines)

### CODE REVIEW

File: docarray/array/storage/base/backend.py [hunk 1]
```diff
@@ -75,7 +75,10 @@ def _map_id(self, _id: str) -> str:
         return _id
 
     def _map_column(self, value, col_type) -> str:
-        return self.TYPE_MAP[col_type].converter(value)
+        if value is not None:
+            return self.TYPE_MAP[col_type].converter(value)
+        else:
+            raise ValueError('Column values should not be None.')
```
Comment (alaeddine-13) [inline docarray/array/storage/base/backend.py:81]: there's no other backend that supports nullable fields ?

Comment (dongxiang123) [reply docarray/array/storage/base/backend.py:81]: Actually other backend also don't support nullbale fields, like None for n_dim in other backends, I think it is better to keep consistent for all the backends.

Comment (JoanFM) [reply docarray/array/storage/base/backend.py:81]: we have to aim to enable `nullable` fields wherever we can

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: docarray/array/storage/base/backend.py (hunk 1 above)
Comment (samsja) [reply docarray/array/storage/base/backend.py:81]: yes I agree I would say allow `nullable` should be the default and we deactivate it for some given backend that don't support it

**Comment by dongxiang123** [comment]
Because we need to switch to tasks with higher priority, this one is temporarily on hold.

**Comment by dongxiang123** [comment]
Here is the summary for backends and whether they are supporting the none value configs:
1. Redis: 
    1. When sending questions, it is not working because dataclass checking.
2. Elastic:
    1. Support by default
3. qdrant:
    1.  not working because of the Qdrantic learning. 
4. annlite: Support by default
5. sqlite:
    1. No founding cases.
6. weaviate: has the room to improve

**Comment by samsja** [comment]
We closed this PR as we are not working on it anymore

**Pull request closed without merging**